### DISCUSSION TITLE
Why mp4 files take a long time so start the very first time?

### DISCUSSION THREAD

**Discussion by juxeii** [photoprism/photoprism#1991] [Q&A] [answered]
Hi, 
I have a Pixel 3a, and videos are saved as *.mp4 file format. 
When these videos are imported to photoprism, they take a long time to open up the first time. I guess, this is because of transcoding taking place?! 
My question is, why is mp4 transcoded at all?
This format should be natively playable in all browsers? 
What can I do to tell photoprism to just "stream" them, instead of transcoding?

**Comment by lastzero** [comment, marked as answer]
https://docs.photoprism.app/getting-started/troubleshooting/#videos-dont-play

**Comment by juxeii** [comment]
Hi,

thx, I know this site.
My problem is not that the videos are not playing, but it takes some time before they play the first time.
After they played the first time, they start in an instant. This is session independent, since they start in an instant days after and on all other devices.
My question is why is mp4 not starting immediately?
Is mp4 so different than mpeg4/avc?
Is there something I can do here to tell photoprism just to stream them?
This site does not answert my questions :(

**Reply by lastzero** [reply]
Just spent my Sunday to further improve our docs and explain it in detail. Check them again! :+1:

**Reply by lastzero** [reply]
https://docs.photoprism.app/user-guide/organize/video/#transcoding

**Comment by juxeii** [comment]
With debug logs enabled, I have now proof that mp4 is transcoded the first time a mp4 file is opened:
_2022-01-30 17:48:07 INFO libx264: transcoding 2021/12/20211202_161229_1BC6B592.mp4 to avc_

So the question remains: why are my mp4 files transcoded at all?

**Reply by juxeii** [reply]
@lastzero Thank you so much for clarification! Would it make sense to run the exiftiool on one on the mp4 files and post the result here? That might show why they are transcoded!?

**Reply by lastzero** [reply]
Yes

**Reply by juxeii** [reply]
I think I nailed it down. Attached are two output files from exiftool
[hvc1.txt](https://github.com/photoprism/photoprism/files/7966630/hvc1.txt)
[avc1.txt](https://github.com/photoprism/photoprism/files/7966631/avc1.txt)
.
My wife messed around with the video codec of her Pixel 3a to "save space" and she activated H.265 in the options.
So the output of exiftool with these options gives files similar to hvc1.txt.
I then changed it to default H.264, and exiftool outputs avc1.txt

Just tested it in Photoprism and everything is working now as expected. No transcoding took place.
Thanks for your support and have a nice sunday :)
